src/windows/process.rs

```diff
@@ -527,6 +512,146 @@ unsafe fn get_cmdline_from_buffer(buffer: *const u16) -> Vec<String> {
+        let result = ReadProcessMemory(
+            handle,
+            pinfo.PebBaseAddress as *mut _,
+            peb.as_mut_ptr() as *mut _,
+            size_of::<PEB>() as SIZE_T,
+            std::ptr::null_mut(),
+        );
+        if !NT_SUCCESS(result) {
+            return PathBuf::new();
+        }
+        let peb = peb.assume_init();
+
+        let mut proc_params = MaybeUninit::<RTL_USER_PROCESS_PARAMETERS>::uninit();
+        let result = ReadProcessMemory(
+            handle,
+            peb.ProcessParameters as *mut PRTL_USER_PROCESS_PARAMETERS as *mut _,
+            proc_params.as_mut_ptr() as *mut _,
+            size_of::<RTL_USER_PROCESS_PARAMETERS>() as SIZE_T,
+            std::ptr::null_mut(),
+        );
+        if !NT_SUCCESS(result) {
+            return PathBuf::new();
+        }
+        let proc_params = proc_params.assume_init();
+
+        let ptr = proc_params.CurrentDirectory.DosPath.Buffer;
+        let size = proc_params.CurrentDirectory.DosPath.Length;
+        (ptr as LPVOID, size as usize)
+    } else {
+        // target is a 32 bit process in wow64 mode
+        use ntapi::ntwow64::{
+            PEB32, PRTL_USER_PROCESS_PARAMETERS32, RTL_USER_PROCESS_PARAMETERS32,
+        };
+
+        let mut peb32 = MaybeUninit::<PEB32>::uninit();
+        let result = ReadProcessMemory(
+            handle,
+            pwow32info,
+            peb32.as_mut_ptr() as *mut _,
+            size_of::<PEB32>() as SIZE_T,
+            std::ptr::null_mut(),
+        );
+        if !NT_SUCCESS(result) {
+            return PathBuf::new();
+        }
+        let peb32 = peb32.assume_init();
+
+        let mut proc_params = MaybeUninit::<RTL_USER_PROCESS_PARAMETERS32>::uninit();
+        let result = ReadProcessMemory(
+            handle,
+            peb32.ProcessParameters as *mut PRTL_USER_PROCESS_PARAMETERS32 as *mut _,
+            proc_params.as_mut_ptr() as *mut _,
+            size_of::<RTL_USER_PROCESS_PARAMETERS32>() as SIZE_T,
+            std::ptr::null_mut(),
+        );
+        if !NT_SUCCESS(result) {
+            return PathBuf::new();
+        }
+        let proc_params = proc_params.assume_init();
+
+        let ptr = proc_params.CurrentDirectory.DosPath.Buffer;
+        let size = proc_params.CurrentDirectory.DosPath.Length;
+        (ptr as LPVOID, size as usize)
+    };
+
+    let mut buffer: Vec<u16> = Vec::with_capacity(size / 2 + 1);
+    buffer.set_len(size / 2);
+    let result = ReadProcessMemory(
+        handle,
+        ptr as *mut _,
+        buffer.as_mut_ptr() as *mut _,
+        size,
+        std::ptr::null_mut(),
+    );
+
+    if !NT_SUCCESS(result) {
+        return PathBuf::new();
+    }
+    buffer.push(0);
+
+    println!(">> {}", wchar_slice_to_string(buffer.as_slice()));
+
+    PathBuf::from(wchar_slice_to_string(buffer.as_slice()))
+}
+
+unsafe fn wchar_slice_to_string(slice: &[u16]) -> String {
+    let mut pos = 0;
+    for x in slice.iter() {
```

Comment:
Instead of this loop, we could actually use [position](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position).

Reply:
Done